pkg/disk/disk.go

```diff
@@ -32,3 +51,90 @@ type Info struct {
 	// Usage is calculated per tenant.
 	Usage uint64
 }
+
+// Performance holds informantion about read and write speed of a disk
+type Performance struct {
+	Path       string  `json:"path"`
+	Error      string  `json:"error,omitempty"`
+	WriteSpeed float64 `json:"writeSpeed"`
+	ReadSpeed  float64 `json:"readSpeed"`
+}
+
+// GetPerformance returns given disk's read and write performance
+func GetPerformance(path string) Performance {
+	perf := Performance{}
+	write, read, err := performance(path)
+	if err != nil {
+		perf.Error = err.Error()
+		return perf
+	}
+	perf.WriteSpeed = write
+	perf.ReadSpeed = read
+	return perf
+}
+
+// Calculate the write and read performance - write and read 10 tmp (1 MiB)
+// files and find the average time taken (Bytes / Sec)
+func performance(fsPath string) (write, read float64, err error) {
+	var count int
+	var totalWriteElapsed time.Duration
+	var totalReadElapsed time.Duration
+
+	randBuf := make([]byte, randBufSize)
+	rand.Read(randBuf)
+
+	// create the enclosing directory
+	err = os.MkdirAll(fsPath, 0777)
+	if err != nil {
+		os.RemoveAll(fsPath)
+		return 0, 0, err
+	}
+
+	for count = 1; count <= perfCount; count++ {
+		buf := bytes.Repeat(randBuf, randParts)
```

Comment:
since we use this buffer repeatedly without change we don't need to do this each and every time in the loop we can just do this once for the entire perfCount() IMO

Reply:
Also perf count can be call `iterations` ?

Reply:
> we can just do this once for the entire perfCount

ah right, done